src/main.rs

```diff
@@ -1,26 +1,27 @@
 use app_background::AppBackground;
-use clock::ClockGadget;
-use cses_status::CSESStatusGadget;
 use utils::{Drawable, Gadget, NetworkRuntime};
-use weather::WeatherGadget;
 
 mod app_background;
+mod gadgets;
 
 pub struct StarboardApp {
     background: AppBackground,
-    clock_gadget: ClockGadget,
-    weather_gadget: WeatherGadget,
-    cses_status_gadget: CSESStatusGadget,
+    gadgets: Vec<Box<dyn Gadget>>,
 }
 
 impl StarboardApp {
     fn new(egui_ctx: &egui::Context) -> Self {
         let network_runtime = setup_network_runtime();
+        let mut gadgets = vec![];
+
+        // FIXME: Also allow users to spawn gadgets as they wish in the UI
+        for gadget_factory in gadgets::GADGET_FACTORIES {
+            gadgets.push(gadget_factory.make_gadget(&network_runtime, egui_ctx));
+        }
+
         Self {
-            background: AppBackground::default(),
-            clock_gadget: ClockGadget::new(&network_runtime, egui_ctx),
-            weather_gadget: WeatherGadget::new(&network_runtime, egui_ctx),
-            cses_status_gadget: CSESStatusGadget::new(&network_runtime, egui_ctx),
+            background: Default::default(),
```

Comment:
This reverts a change made in PR #7, use `AppBackground::default()` instead

Reply:
Merges are my enemy lol